### ISSUE TITLE
"Cannot read property 'length' of null"

### ISSUE THREAD

**Issue by CactiChameleon9** [ilyakooo0/warden#1]
When trying to sign in with my work account (on the offical instance), I get the error "Cannot read property 'length' of null" with a white loading screen below.

Log:
```
Creating a QMirClientScreen now
Started as a local application container.
qml: Loading webview on /home/phablet/.local/share/soy.iko.warden
Sandboxing disabled by user.
[9472:9503:0814/080713.061320:ERROR:file_path_watcher_linux.cc(71)] Failed to read /proc/sys/fs/inotify/max_user_watches
qq= WebViewImplOxide_QMLTYPE_108(0x142d26c0) 0x14413720
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onContextMenuIntercepted"
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onInsertionHandleTapped"
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onStatusChanged"
qml: Start browsing
QDir::mkpath: Empty or null file name
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:145:9: Unable to assign [undefined] to QObject*
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.126\" (uid=32011 pid=9472 comm=\"webapp-container --local-content-can-access-remote\") interface=\"com.canonical.libertine.Service.Operations\" member=\"list\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=6050 comm=\"/usr/bin/python3 /usr/bin/libertined \")"                        
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.126\" (uid=32011 pid=9472 comm=\"webapp-container --local-content-can-access-remote\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"running\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=6050 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.126\" (uid=32011 pid=9472 comm=\"webapp-container --local-content-can-access-remote\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"last_error\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=6050 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.126\" (uid=32011 pid=9472 comm=\"webapp-container --local-content-can-access-remote\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"result\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=6050 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
[9472:9515:0814/080713.280674:ERROR:zygote_host_impl_linux.cc(263)] Failed to adjust OOM score of renderer with pid 9516: Permission denied (13)
qml: Loaded 14 UA override(s) from file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/ua-overrides-mobile.js
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebContext.qml:104: Error: Invalid write to global property "userAgentOverrides"
qml: user agent: Mozilla/5.0 (Linux; Ubuntu 16.04 like Android 9) AppleWebKit/537.36 Chrome/87.0.4280.144 Mobile Safari/537.36
qml: [JS] (file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1) {"tag":"Init","contents":[]}
qml: [JS] (file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1) {"tag":"Login","contents":{"email":"emailemailemail","password":"","server":"https://vault.bitwarden.com"}}
[9472:9505:0814/080737.461546:ERROR:nss_util.cc(166)] Error initializing NSS with a persistent database (sql:/home/phablet/.pki/nssdb): NSS error code: -8126
qml: [JS] (file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1) TypeError: Cannot read property 'length' of null
    at show (file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1709889)
    at file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1711417
    at file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1711367
    at file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1711367
    at file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1711367
    at file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1711367
    at file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1711367
    at file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1711367
    at file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1711367
    at show (file:///opt/click.ubuntu.com/soy.iko.warden/1.0.0/backend.js:1:1710445)
QObject::startTimer: Timers cannot be started from another thread
```

**Comment by CactiChameleon9** [comment]
![screenshot20220814_080508936](https://user-images.githubusercontent.com/51231053/184526299-4b09416b-7446-4b47-aba7-5ddd251fb297.png)

**Comment by ilyakooo0** [comment]
Does your work account require 2FA?

**Comment by CactiChameleon9** [comment]
I don't think so

**Comment by CactiChameleon9** [comment]
@ilyakooo0 would you mind deleating your edit, I leaked my email in logs 😅

**Comment by ilyakooo0** [comment]
@CactiChameleon9 Done

I'm thinking of disabling logs in production builds entirely, since it's pretty much useless without private data

**Comment by CactiChameleon9** [comment]
Thanks, sorry about that 😂. Is there anything I can do to help debug this?

**Comment by ilyakooo0** [comment]
This issue is related to Bitwarden having basically everything as nullable without any documentation as to what can actually be returned.

It is most likely that one of the few strings I didn't declare as nullable being `null`.

The ideal case would be to see the encrypted contents of the vault (the structure itself is visible, but all strings are encrypted with a master password, so nothing is really leaked), but I don't think you would be comfortable sharing that ;D

I will create a patch and ask you to test the patched version.

**Comment by ilyakooo0** [comment]
@CactiChameleon9 could you try the build from this commit: https://github.com/ilyakooo0/warden/actions/runs/2855229883 

(see "Artifacts" at the bottom of the page)

**Comment by CactiChameleon9** [comment]
I have installed, but I still get the error

**Comment by ilyakooo0** [comment]
@CactiChameleon9 what about this one? (: https://github.com/ilyakooo0/warden/actions/runs/2856567449

**Comment by ilyakooo0** [comment]
@CactiChameleon9 what about this one? (: https://github.com/ilyakooo0/warden/actions/runs/2856567449

**Comment by CactiChameleon9** [comment]
Still no!!! @ilyakooo0

**Comment by ilyakooo0** [comment]
@CactiChameleon9  https://github.com/ilyakooo0/warden/actions/runs/2859488816

If this one doesn't work, then I don't know what's wrong

**Comment by CactiChameleon9** [comment]
@ilyakooo0 It works!!! Thanks a lot for trying to solve this

**Comment by alex-startengine** [comment]
can you release a new version for ubports stable branch with this fix?

**Comment by ilyakooo0** [comment]
@alex-startengine released version 1.1.0
